src/libraries/System.Net.Sockets/src/System/Net/Sockets/UdpReceiveResult.cs

```diff
@@ -71,12 +71,12 @@ public override int GetHashCode()
         /// <returns>true if obj is an instance of <see cref="UdpReceiveResult"/> and equals the value of the instance; otherwise, false</returns>
         public override bool Equals(object obj)
         {
-            if (!(obj is UdpReceiveResult))
+            if (!(obj is UdpReceiveResult other))
             {
                 return false;
             }
 
-            return Equals((UdpReceiveResult)obj);
+            return Equals(other);
```

Comment:
Nit: as long as you're changing this, you could change it to the more idiomatic:
```C#
public override bool Equals(object obj) => object is UdpReceiveResult other && Equals(other);
```